libwallet/assets_manager.go

```diff
@@ -960,3 +961,26 @@ func (mgr *AssetsManager) DeleteDEXData() error {
 	// Delete dex client db.
 	return os.Remove(dexDBFile)
 }
+
+func (mgr *AssetsManager) ListenAssetChange(listen func()) {
+	// Reload wallets unmixed balance and reload UI on new blocks.
+	txAndBlockNotificationListener := &sharedW.TxAndBlockNotificationListener{
+		OnTransactionConfirmed: func(_ int, _ string, _ int32) {
+			listen()
+		},
+		OnTransaction: func(_ int, _ *sharedW.Transaction) {
+			listen()
+		},
+	}
+	for _, wallet := range mgr.AllWallets() {
+		if err := wallet.AddTxAndBlockNotificationListener(txAndBlockNotificationListener, assetID); err != nil {
+			log.Errorf("Can't listen tx and block notification for %s wallet", wallet.GetWalletName())
```

Comment:
this error can also be as a result of the notification listener already added correct?, in that case this is prone to spam the logs.

maybe check if tx and block notification listener isn't nil before adding it